x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_pipelines.test.ts

```diff
@@ -0,0 +1,131 @@
+  const mockClient = {
+    ingest: {
+      getPipeline: jest.fn(),
+    },
+  };
+  const mockTrainedModelsProvider = {
+    getTrainedModels: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should throw an error if Machine Learning is disabled in the current space', () => {
+    expect(() =>
+      getMlInferencePipelines(mockClient as unknown as ElasticsearchClient, undefined)
+    ).rejects.toThrowError('Machine Learning is not enabled');
+  });
+
+  it('should fetch inference pipelines and redact inaccessible model IDs', async () => {
+    function mockInferencePipeline(modelId: string) {
+      return {
+        processors: [
+          {
+            append: {},
+          },
+          {
+            inference: {
+              model_id: modelId,
+            },
+          },
+          {
+            remove: {},
+          },
+        ],
+      };
+    }
+
+    const mockPipelines = {
+      pipeline1: mockInferencePipeline('model1'),
+      pipeline2: mockInferencePipeline('model2'),
+      pipeline3: mockInferencePipeline('redactedModel3'),
+      pipeline4: {
+        // Pipeline with multiple inference processors referencing an inaccessible model
+        processors: [
+          {
+            append: {},
+          },
+          {
+            inference: {
+              model_id: 'redactedModel3',
+            },
+          },
+          {
+            inference: {
+              model_id: 'model2',
+            },
+          },
+          {
+            inference: {
+              model_id: 'redactedModel4',
+            },
+          },
+          {
+            remove: {},
+          },
+        ],
+      },
+    };
+
+    const mockTrainedModels = {
+      trained_model_configs: [
+        {
+          model_id: 'model1',
+        },
+        {
+          model_id: 'model2',
+        },
+      ],
+    };
+
+    mockClient.ingest.getPipeline.mockImplementation(() => Promise.resolve(mockPipelines));
+    mockTrainedModelsProvider.getTrainedModels.mockImplementation(() =>
+      Promise.resolve(mockTrainedModels)
+    );
+
+    const actualPipelines = await getMlInferencePipelines(
+      mockClient as unknown as ElasticsearchClient,
+      mockTrainedModelsProvider as unknown as MlTrainedModels
+    );
+
+    expect(
+      (actualPipelines.pipeline1.processors as IngestProcessorContainer[])[1].inference?.model_id
+    ).toBeDefined();
+    expect(
+      (actualPipelines.pipeline2.processors as IngestProcessorContainer[])[1].inference?.model_id
+    ).toBeDefined();
+    expect(
+      (actualPipelines.pipeline3.processors as IngestProcessorContainer[])[1].inference?.model_id
+    ).toEqual(''); // Redacted model ID
+    expect(
+      (actualPipelines.pipeline4.processors as IngestProcessorContainer[])[1].inference?.model_id
+    ).toEqual('');
+    expect(
+      (actualPipelines.pipeline4.processors as IngestProcessorContainer[])[2].inference?.model_id
+    ).toBeDefined();
```

Comment:
Does `''` pass the `toBeDefined()` check?

Reply:
Actually it does. So we're not proving that we're getting an exact model ID (and not disproving the possible bug that _every_ model ID gets redacted). These lines should say something like `.toEqual('model1')`.
Let me fix that in a follow-up PR as this one is already approved and good to go.

Reply:
Addressed in #143967.